redbpf-probes/src/maps.rs

```diff
@@ -392,6 +392,150 @@ impl<T> PerfMap<T> {
     }
 }
 
+#[cfg(feature = "ringbuf")]
+mod ringbuf {
+    use super::*;
+
+    /// Flags that be passed to `RingBuf::output_with_flags`.
+    #[derive(Debug, Copy, Clone)]
+    pub struct RingBufMapFlags(u32);
+
+    impl Default for RingBufMapFlags {
+        #[inline]
+        fn default() -> Self {
+            RingBufMapFlags(0)
+        }
+    }
+
+    impl RingBufMapFlags {
+        /// Create new default flags.
+        #[inline]
+        pub fn new() -> Self {
+            Default::default()
+        }
+
+        /// Set BPF_RB_NO_WAKEUP flag.
+        #[inline]
+        pub fn no_wakeup(&mut self) -> &mut RingBufMapFlags {
+            self.0 |= BPF_RB_NO_WAKEUP;
+            self
+        }
+
+        /// Set BPF_RB_FORCE_WAKEUP flag
+        #[inline]
+        pub fn force_wakeup(&mut self) -> &mut RingBufMapFlags {
+            self.0 |= BPF_RB_FORCE_WAKEUP;
+            self
+        }
+    }
+
+    /// Ring buffer map.
+    ///
+    /// This is a wrapper for `BPF_MAP_TYPE_RINGBUF`.
+    pub struct RingBufMap<T> {
+        def: bpf_map_def,
+        _event: PhantomData<T>,
+    }
+
+    impl<T> RingBufMap<T> {
+        /// Creates a ring buffer map with the specied maximum number of elements.
+        ///
+        /// `max_entries` must be a power of 2 value and must be page aligned.
+        pub const fn with_max_entries(max_entries: u32) -> Self {
```

Comment:
`max_entries` keeps it consistent with other maps of the like, so I think it's preferable.

Reply:
I chose `max_entries` for consistency, but really this is allocating a buffer of a certain size rather than an array of elements. `with_buffer_size` is more descriptive.

Reply:
Changed the name and docs. I think the docs better match the behaviour now.